portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,23 +23,40 @@
 import javax.servlet.http.HttpServletResponse;
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> fruits;
+  private ArrayList<String> comments;
+
 
   @Override
   public void init(){
-    fruits = new ArrayList<String>();
-    fruits.add("Blueberry");
-    fruits.add("Cherry");
-    fruits.add("Peach");
-    fruits.add("Apricot");
-    fruits.add("Mango");
+    comments = new ArrayList<String>();    
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String jsonfruit = gson.toJson(fruits);
     response.setContentType("application/json;");
-    response.getWriter().println(jsonfruit);
+    String json = new Gson().toJson(comments);
+    response.getWriter().println(json);
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String comment = getComment(request);
+      comments.add(comment);
+      response.setContentType("text/html");
+      response.sendRedirect("/index.html");
+  }
+
+  /**
```

Comment:
If you're going to include a javadoc for a function (unlike in javascript, javadocs are not required for private functions, but are always nice to include if they help the codes readability), make sure you include all of the applicable tags (@param, @return, @throws and/or @deprecated).  In this case, make sure you add a param tag for the request variable.

Reply:
Noted! Thanks for including those other tags, I was only aware of @return and @param